src/Language/PureScript/Sugar/DoNotation.hs

```diff
@@ -40,6 +42,13 @@ desugarDo d =
   replace _ (PositionedValue pos com v) = PositionedValue pos com <$> rethrowWithPosition pos (replace pos v)
   replace _ other = return other
 
+  stripPositionedBinder :: Binder -> (Maybe SourceSpan, Binder)
+  stripPositionedBinder (PositionedBinder ss _ b) =
+    let (ss', b') = stripPositionedBinder b
+     in (ss' <|> Just ss, b')
+  stripPositionedBinder b =
+    (Nothing, b)
+
```

Comment:
@hdgarrood Is this sort of check going to need to be done for `ado` as well?

Reply:
If we want to preserve variable names from the source in the output rather than using generated names, yes. I think I'd prefer to tackle that separately though.